File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).


% `G` -> `C`
% `C` -> `G`
% `T` -> `A`
% `A` -> `U`

to_rna(Strand) -> 
    lists:map(
      fun(S) ->
          case S of
              $G -> $C;
              $C -> $G;
              $T -> $A;
              $A -> $U
          end
      end, Strand).

File: c/rna-transcription/rna_transcription.c
#include <stdlib.h>
#include <string.h>

#include "rna_transcription.h"

char *to_rna(const char *dna) {
    size_t len = strlen(dna);
    char *rna = malloc(len+1);
    
    for (size_t i = 0; i < len; i++) {
        switch (dna[i]) {
            case 'G': rna[i] = 'C'; break;
            case 'C': rna[i] = 'G'; break;
            case 'T': rna[i] = 'A'; break;
            case 'A': rna[i] = 'U'; break;
            default: rna[i] = '\0';
        }
    }
    rna[len] = '\0';
    return rna;
}

